superset-frontend/src/views/CRUD/data/dataset/AddDataset/Footer/index.tsx

```diff
@@ -17,7 +17,103 @@
+import { useSingleViewResource } from 'src/views/CRUD/hooks';
+import { logEvent } from 'src/logger/actions';
+import withToasts from 'src/components/MessageToasts/withToasts';
+import {
+  LOG_ACTIONS_DATASET_CREATION_EMPTY_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_DATABASE_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_SCHEMA_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_TABLE_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_SUCCESS,
+} from 'src/logger/LogUtils';
+import { DatasetObject } from '../types';
 
-export default function Footer() {
-  return <div>Footer</div>;
+const INPUT_FIELDS = ['db', 'schema', 'table_name'];
+const LOG_ACTIONS = [
+  LOG_ACTIONS_DATASET_CREATION_EMPTY_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_DATABASE_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_SCHEMA_CANCELLATION,
+  LOG_ACTIONS_DATASET_CREATION_TABLE_CANCELLATION,
+];
+interface FooterProps {
+  url: string;
+  addDangerToast: () => void;
+  datasetObject?: Partial<DatasetObject> | null;
+  onDatasetAdd?: (dataset: DatasetObject) => void;
 }
+
+function Footer({ url, datasetObject, addDangerToast }: FooterProps) {
+  const { createResource } = useSingleViewResource<Partial<DatasetObject>>(
+    'dataset',
+    t('dataset'),
+    addDangerToast,
+  );
+
+  const createLogAction = (dataset: Partial<DatasetObject>) => {
+    let totalCount = 0;
+    const value = Object.keys(dataset).reduce((total, key) => {
+      if (INPUT_FIELDS.includes(key) && dataset[key]) {
+        totalCount += 1;
+      }
+      return totalCount;
+    }, 0);
+
+    return LOG_ACTIONS[value];
+  };
+  const goToPreviousUrl = () => {
+    // this is a placeholder url until the final feature gets implemented
+    // at that point we will be passing in the url of the previous location.
+    window.location.href = url;
+  };
+
+  const cancelButtonOnClick = () => {
+    if (!datasetObject) {
+      logEvent(LOG_ACTIONS_DATASET_CREATION_EMPTY_CANCELLATION, {});
+    } else {
+      const logAction = createLogAction(datasetObject);
+      logEvent(logAction, datasetObject);
+    }
+    goToPreviousUrl();
+  };
+
+  const tooltipText = t('Select a database table.');
+
+  const onSave = () => {
+    if (datasetObject) {
+      const data = {
+        database: datasetObject.db?.id,
+        schema: datasetObject.schema,
+        table_name: datasetObject.table_name,
+      };
+      createResource(data).then(response => {
+        if (!response) {
+          return;
+        }
+        if (typeof response === 'number') {
+          logEvent(LOG_ACTIONS_DATASET_CREATION_SUCCESS, datasetObject);
+          // When a dataset is created the response we get is its ID number
+          goToPreviousUrl();
+        }
+      });
+    }
+  };
+
+  return (
+    <>
+      <Button onClick={cancelButtonOnClick}>Cancel</Button>
+      <Button
+        buttonStyle="primary"
+        disabled={!datasetObject?.table_name}
+        tooltip={!datasetObject?.table_name ? tooltipText : undefined}
```

Comment:
can the tooltip take other falsy values or does it require undefined?

Reply:
The type wants undefined, I originally tried it with null but got ts errors.

Reply:
ok, thanks!